libcaf_test/caf/test/fixture/deterministic_logger.hpp

```diff
@@ -0,0 +1,64 @@
+#include "caf/logger.hpp"
+
+using namespace caf;
+
+namespace caf::test::fixture {
+
+/// A logger implementation for deterministic fixture to log into console
+class deterministic_logger : public caf::logger,
+                             public caf::detail::atomic_ref_counted {
```

Comment:
This class doesn't need to be exposed to the public API. Hence, it should move to `libcaf_test/caf/test/fixture/deterministic.cpp` as an implementation detail.